deploy/crds/mongodb.com_v1_mongodb_cr.yaml

```diff
@@ -12,11 +13,22 @@ spec:
   users:
     - name: my-user
       db: admin
-      passwordSecretRef:
+      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
         name: my-user-password
       roles:
         - name: clusterAdmin
           db: admin
         - name: userAdminAnyDatabase
           db: admin
       scramCredentialsSecretName: my-scram
+
+# the user credentials will be generated from this secret
```

Comment:
q: did we consider the operator deleting it(automatically) after the credentials are generated?

Reply:
yes, we decided against having the operator delete user created resources as this could end up being an unexpected behaviour. This secret might be intended for use elsewhere as well.